subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/IndexedNormalizedFileSnapshot.java

```diff
@@ -38,4 +38,32 @@ public String getAbsolutePath() {
     public int getIndex() {
         return index;
     }
+
+    @Override
+    protected boolean hasSameNormalizedPathAs(AbstractNormalizedFileSnapshot that) {
+        if (that instanceof IndexedNormalizedFileSnapshot) {
```

Comment:
does it make sense to test for reference equality first?

Reply:
Reference equality of what? The super method already checks equality to the other snapshot.

Reply:
Ok I see. It's bit contrived, because this is _almost-a_ equals method :)